bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/ghost/GhostEditor.scala

```diff
@@ -56,87 +59,99 @@ final class GhostEditor extends ComponentEditor[ISPObsComponent, Ghost] {
     layout(asterismLabel) = new Constraints() {
       anchor = Anchor.East
-      gridwidth = 2
-      gridy = 3
-      insets = new Insets(22, 10, 0, 20)
+      gridy = 2
+      insets = new Insets(12, 10, 0, 20)
     }
 
-    // TODO: This will change. I'm going to make it call transformation methods between asterism types.
-    // TODO: I still have a little thinking as to how this will be implemented and where these names
-    // TODO: belong, so this is just a "placeholder" for now.
-    val asterismMap: Map[String, () => GhostStandardResTargets] = Map(
-      "Single target" -> (() => GhostStandardResTargets.emptySingleTarget),
-      "Dual target"   -> (() => GhostStandardResTargets.emptyDualTarget),
-      "SRIFU1 target, SRIFU2 sky position" -> (() => GhostStandardResTargets.emptyTargetPlusSky),
-      "SRIFU1 sky position, SRIFU2 target" -> (() => GhostStandardResTargets.emptySkyPlusTarget)
+    /** A map between GHOST asterism names and converters to those types. */
+    val asterismMap: Map[String, GhostAsterismConverter] = ListMap(
+      "Single target" -> GhostSingleTargetConverter,
+      "Dual target" -> GhostDualTargetConverter,
+      "SRIFU1 target, SRIFU2 sky position" -> GhostTargetPlusSkyConverter,
+      "SRIFU1 sky position, SRIFU2 target" -> GhostSkyPlusTargetConverter,
+      "High resolution" -> GhostHighResolutionConverter
     )
 
     val asterismComboBox: ComboBox[String] = new ComboBox[String](asterismMap.keys.toSeq)
     layout(asterismComboBox) = new Constraints() {
       anchor = Anchor.NorthWest
-      gridx = 2
-      gridy = 3
-      insets = new Insets(20, 0, 0, 20)
+      gridx = 1
+      gridwidth = 2
+      gridy = 2
+      insets = new Insets(10, 0, 0, 20)
     }
 
     layout(new Label) = new Constraints() {
       anchor = Anchor.North
-      gridy = 4
+      gridy = 3
       weighty = 1.0
     }
 
-    listenTo(standardResolution, highResolution, asterismComboBox)
+    listenTo(asterismComboBox.selection)
     reactions += {
-      case ButtonClicked(`standardResolution`) =>
-        showStandardResolutionComponents(true)
-      case ButtonClicked(`highResolution`) =>
-        showStandardResolutionComponents(false)
+      case SelectionChanged(`asterismComboBox`) =>
+        val converter = asterismMap(asterismComboBox.selection.item)
+        convertAsterism(converter)
     }
 
-    def initialize(): Unit = {
-      val isHighResolution = Option(getContextTargetEnv).exists(_.getAsterism.isInstanceOf[GhostAsterism.HighResolution])
-      if (isHighResolution)
-        highResolution.doClick()
-      else
-        standardResolution.doClick()
+    /** Convert the asterism to the new type, and set the new target environment. */
+    def convertAsterism(converter: GhostAsterismConverter): Unit = Swing.onEDT {
+      val o = getContextObservation
+      val oc = getContextTargetObsComp
+
+      val newToc = for {
+        toc <- o.findTargetObsComp
```

Comment:
If you've got `oc` above (which could be `null` so you should check for that) then you can just ask it for its data object:  

```scala
val action = for {
  oc  <- Option(getContextTargetObsComp)
  toc <- Option(oc.getDataObject).collect { case t: TargetObsComp => t }`
  env <- converter.convert(toc.getTargetEnvironment)
} yield {
  // Here I want something like cats.Eval.later() but since we don't have that
  () => { toc.setTargetEnvironment(env); oc.setDataObject(toc) }
}
```

This assumes you want to show the dialog box if something goes wrong.  You'd apply the action if it is defined or else show the dialog.  I'm wondering about this though.  Why can the conversion fail?  Is it just for the case where you're dealing with a `SingleTargetAsterism` that shouldn't be there?  If so that case should probably just disable the combobox in the first place (and show a Phase 2 check error), in which case you wouldn't `yield` anything above and instead just apply the action directly.